### DISCUSSION TITLE
Is microcks compatible with Azure Cosmos DB for MongoDB

### DISCUSSION THREAD

**Discussion by suic86** [microcks/microcks#883] [Q&A]
Is microcks compatible with Azure Cosmos DB for MongoDB? I've to set up Cosmos DB as backend for microcks, but I've been constantly getting "BadValue" (https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/error-codes-solutions#:~:text=2,query%20being%20attempted.) error due to indexing.

I run microcks on an AKS cluster using a modified version of the provided helm chart. microcks works fine when I run it with MongoDB running as a pod. microcks also works with MongoDB Atlas. It looks like, that the issue is caused by the differences between MongoDB and Cosmos DB for MongoDB which is technically not a MongoDB (https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/error-codes-solutions#:~:text=Azure%20Cosmos%20DB%20does%20not%20host%20the%20MongoDB%20engine.%20It%20provides%20an%20implementation%20of%20the%20MongoDB%20wire%20protocol%20version%204.0%2C%203.6%2C%20and%20legacy%20support%20for%20wire%20protocol%20version%203.2.%20Therefore%2C%20some%20of%20these%20errors%20are%20only%20found%20in%20Azure%20Cosmos%20DB%27s%20API%20for%20MongoDB.). Therefore, some of these errors are only found in Azure Cosmos DB's API for MongoDB.)

If Cosmos DB for MongoDB is not supported, if the support planned in a future releases? Should I raise a feature request for this?

Thanks

**Comment by lbroudoux** [comment]
Hello! Sorry for the delay in answering but summer time vacation 😉 

We didn't test Microcks with Cosmos DB or even heard about someone in the community who did it. Regarding the error you're facing, I'd say that you may try the `nightly` build of Microcks that is based on branch `1.8.x`. We recently changed the MongoDB driver here and had to upgrade the wire protocol to 3.6. So it may make things better 🤞 

Let us know how things are going and then - yes, it may makes sense to raise a feature request.

Cheers,
